shared/api/c_api_tokenizer.cc

```diff
@@ -373,7 +436,18 @@ extError_t ORTX_API_CALL OrtxDetokenizeCached(const OrtxTokenizer* tokenizer, Or
   }
 
   cache_ptr->last_text_.clear();
-  status = ReturnableStatus(token_ptr->Id2Token(next_id, cache_ptr->last_text_, cache_ptr->decoder_state_));
+
+  // If skip_special_tokens option exists, use its value, otherwise use default (true)
+  auto it = token_ptr->options_map.find("skip_special_tokens");
+  if (it != token_ptr->options_map.end()) {
+    bool skip_special_tokens = it->second == "true";
```

Comment:
I see that some places the option value can accept "false" as well as "0". Do we then also want to support "1". You could abstract this into a function of its own like 

```
bool skip_special_tokens = ParseOption<bool>(option_map, SkipSpecialTokensOptionName)
```

Reply:
Great catch! So the default for `skip_special_tokens` is actually `true` so it suffices to check for "false" or "0" like I did above for `add_special_tokens` in [lines 168-173 in this file](https://github.com/microsoft/onnxruntime-extensions/pull/998/files/570f3130746d5e2f9c58527e25089b78c6ecb22c#diff-653e4ccf47f2ec04ce53e9eb92941686a5c922d5d5ddffbe9acb464ee8013bd5R171), but I missed doing that here - fixed in the latest commit.

As for `ParseOption`, in general I completely agree we should modularize logic like this as much as possible, but in this case, I think it is not needed given how straightforward it is - I'd like to keep the C API as clean as possible. We can do this in the future when we support many more options however.

Reply:
done!